pkgs/by-name/sp/spades/package.nix

```
@@ -1,40 +1,54 @@
 {
   lib,
   stdenv,
-  fetchurl,
-  zlib,
   bzip2,
   cmake,
+  fetchurl,
+  fetchpatch,
+  ncurses,
   python3,
+  readline,
+  zlib-ng,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "spades";
-  version = "3.15.5";
+  version = "4.0.0";
 
   src = fetchurl {
-    url = "https://github.com/ablab/spades/releases/download/v${finalAttrs.version}/SPAdes-${finalAttrs.version}.tar.gz";
-    hash = "sha256-FVw2QNVx8uexmgUDHR/Q0ZvYLfeF04hw+5O9JBsSu/o=";
+    url =
+      "https://github.com/ablab/spades/releases/download/v${finalAttrs.version}/SPAdes-${finalAttrs.version}.tar.gz";
+    hash = "sha256-B8AusdnZD2EaxzvdMN3CQu1RsAyKN1cYnoqBN62M+4s=";
   };
   sourceRoot = "SPAdes-${finalAttrs.version}/src";
 
-  env.CXXFLAGS = toString [
-    # GCC 13: error: 'uint32_t' does not name a type
-    "-include cstdint"
+  patches = [
+    (fetchpatch {
+      name = "copytree.patch";
+      url = "https://github.com/ablab/spades/pull/1314/commits/af9feaf4cc7cbd9102362309f582e8645c471e02.patch";
+      sha256 = "sha256-tkT7hb0TqsbLkcTs9u43nzvV8bVdh3G9VKYqFFLrQv8=";
```

```suggestion
      hash = "sha256-tkT7hb0TqsbLkcTs9u43nzvV8bVdh3G9VKYqFFLrQv8=";
```
Prefer `hash` over `sha256`.